plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotIntegrationSmokeTest.java

```diff
@@ -0,0 +1,322 @@
+            if (i % 2 == 0) {
+                stringList.add("string_" + abs(random.nextInt(10001)));
+            }
+            else {
+                stringList.add(null);
+            }
+        }
+        List<Boolean> booleanList = new ArrayList<>();
+        for (int i = 0; i < 5; i++) {
+            if (i % 2 == 0) {
+                booleanList.add(random.nextBoolean());
+            }
+            else {
+                booleanList.add(null);
+            }
+        }
+
+        List<Integer> integerList = new ArrayList<>();
+        for (int i = 0; i < 5; i++) {
+            integerList.add(null);
+        }
+
+        List<Integer> integerWithDefaultList = new ArrayList<>();
+        for (int i = 0; i < 5; i++) {
+            if (i % 2 == 0) {
+                integerWithDefaultList.add(random.nextInt(10000));
+            }
+            else {
+                integerWithDefaultList.add(null);
+            }
+        }
+
+        List<Float> floatList = new ArrayList<>();
+        floatList.add(null);
+
+        List<Integer> doubleList = new ArrayList<>();
+        doubleList.add(null);
+
+        return new GenericRecordBuilder(schema)
+                .set("string_array_col", stringList)
+                .set("bool_array_col", booleanList)
+                .set("int_array_col", integerList)
+                .set("int_array_col_with_pinot_default", integerWithDefaultList)
+                .set("float_array_col", floatList)
+                .set("double_array_col", doubleList)
+                .set("long_array_col", new ArrayList<>())
+                .build();
+    }
+
+    private static GenericRecord createTestRecord(
+            List<String> stringArrayColumn,
+            List<Boolean> booleanArrayColumn,
+            List<Integer> intArrayColumn,
+            List<Float> floatArrayColumn,
+            List<Double> doubleArrayColumn,
+            List<Long> longArrayColumn,
+            long updatedAtMillis)
+    {
+        Schema schema = getAvroSchema();
+
+        return new GenericRecordBuilder(schema)
+                .set("string_CoL", stringArrayColumn.get(0))
+                .set("bool_COL", booleanArrayColumn.get(0))
+                .set("bytes_col", Hex.toHexString(stringArrayColumn.get(0).getBytes(StandardCharsets.UTF_8)))
+                .set("string_array_col", stringArrayColumn)
+                .set("bool_array_col", booleanArrayColumn)
+                .set("int_array_col", intArrayColumn)
+                .set("int_array_col_with_pinot_default", intArrayColumn)
+                .set("float_array_col", floatArrayColumn)
+                .set("double_array_col", doubleArrayColumn)
+                .set("long_array_col", longArrayColumn)
+                .set("int_CoL", intArrayColumn.get(0))
+                .set("float_col", floatArrayColumn.get(0))
+                .set("double_col", doubleArrayColumn.get(0))
+                .set("long_CoL", longArrayColumn.get(0))
```

Comment:
Some names are lowercase, some are not.
Maybe we could have all lowercase here and have a dedicated test for non-lowercase name handling? (see TestPostgreSqlCaseInsensitiveMapping for example)